circuit-lib/tests/elgamal.test.ts

```diff
@@ -0,0 +1,272 @@
+      encryption2.encrypted_message,
+    );
+    const ephemeral_key3 = encryption1.ephemeral_key.add(
+      encryption2.ephemeral_key,
+    );
+
+    const decrypted_message3 = decrypt(
+      keypair.privKey,
+      ephemeral_key3,
+      encrypted_message3,
+    );
+
+    expect(
+      decrypted_message3.toAffine(),
+      "Invalid linear homomorphism!",
+    ).to.deep.equal(message3.toAffine());
+  });
+
+  it("Check unhappy homomorphic properties for wrong inputs", () => {
+    const keypair = genKeypair();
+
+    const encryption1 = encrypt(keypair.pubKey);
+    const encryption2 = encrypt(keypair.pubKey);
+
+    const message3 = encryption1.message.add(encryption2.message);
+    const encrypted_message3 = encryption1.encrypted_message.add(
+      encryption2.encrypted_message,
+    );
+    // we only modifiy ephemeral_key3 in this example
+    const ephemeral_key3 = encryption1.ephemeral_key.add(babyJub.BASE);
+
+    const decrypted_message3 = decrypt(
+      keypair.privKey,
+      ephemeral_key3,
+      encrypted_message3,
+    );
+
+    expect(
+      decrypted_message3.toAffine(),
+      "Invalid linear homomorphism!",
+    ).to.not.deep.equal(message3.toAffine());
+  });
+});
+
+describe("Testing Encoding/Decoding for ElGamal Scheme", async () => {
+  let lookupTable;
+  const lookupTable19Path = `./src/elgamal-babyjub/lookupTableBBJub19.json`;
+  before(() => {
+    if (!fs.existsSync(lookupTable19Path)) {
+      console.log("Building the lookup table file...");
+      precompute(19);
+      lookupTable = JSON.parse(fs.readFileSync(lookupTable19Path));
+    } else lookupTable = JSON.parse(fs.readFileSync(lookupTable19Path));
+  });
+
+  it("Check encoding a plain text bigger than 32 bits returns error", () => {
+    const plaintext = 4294967297n;
+    let expected = Error;
+    const exercise = () => encode(plaintext);
+    assert.throws(exercise, expected);
+  });
+
+  it("Check encoded value is a valid BabyJub point", () => {
+    const plaintext = pruneTo32Bits(genRandomSalt());
+    const encoded = encode(plaintext);
+    encoded.assertValidity();
+  });
+
+  it("Check compliance of orignal and decoded message as 32-bit numbers", async () => {
+    const plaintext = pruneTo32Bits(genRandomSalt());
+    const encoded = encode(plaintext);
+    const decoded = decode(encoded, 19, lookupTable);
+
+    assert(plaintext === decoded, "Decoded number is different!");
+  });
+
+  it("Check LOOPED compliance of orignal and decoded message as 32-bit numbers", async () => {
+    for (let i = 0; i < 15; i++) {
+      let plaintext = pruneTo32Bits(genRandomSalt());
+      let encoded = encode(plaintext);
+      let decoded = decode(encoded, 19, lookupTable);
+
+      assert(plaintext === decoded, "Decoded number is different!");
+    }
+  });
+
+  it("Check decoding preserves Elgamal linear homomorphism", async () => {
+    // The input should be a 64-bit number
+    const plaintext = pruneTo64Bits(genRandomSalt());
```

Comment:
you could also just get 8 random bytes

Reply:
replaced by `getRandomBytes()` function for both 32 and 64-bit numbers.